src/mame/sinclair/scorpion.cpp

```diff
@@ -201,21 +300,27 @@ void scorpion_state::machine_start()
 	spectrum_128_state::machine_start();
 
 	save_item(NAME(m_port_1ffd_data));
+	save_item(NAME(m_nmi_pending));
+	save_item(NAME(m_ay_selected));
+	save_item(NAME(m_ram_banks));
+
+	m_maincpu->space(AS_PROGRAM).specific(m_program);
```

Comment:
Can you also get a `memory_access<...>::specific` for `m_bankio`?  Since it implements `device_memory_interface`, it should be possible to call `m_bankio->space(0).specific(...);` so you can avoid using the `read8` and `write8` trampolines.  Since you aren’t actually using the banking functionality, the logic in `read8` and `write8` can be ignored.